openshift/Makefile

```diff
@@ -1,15 +1,19 @@
 BIN_DIR := bin
-TOOLS_DIR := hack/tools
-TOOLS_BIN_DIR := $(TOOLS_DIR)/$(BIN_DIR)
-KUSTOMIZE := ../$(TOOLS_BIN_DIR)/kustomize
+TOOLS_DIR := tools
 
 $(RELEASE_DIR):
 	mkdir -p $(RELEASE_DIR)/
 
-$(KUSTOMIZE): # Download kustomize using hack script into tools folder.
-	../hack/ensure-kustomize.sh
+KUSTOMIZE := tools/bin/kustomize
+MANIFESTS_GEN := go run ./vendor/github.com/openshift/cluster-capi-operator/manifests-gen/
+
+$(KUSTOMIZE):
+	./tools/ensure-kustomize.sh
 
 .PHONY: ocp-manifests
 ocp-manifests: $(RELEASE_DIR) $(KUSTOMIZE) ## Builds openshift specific manifests
 	# Build core-components.
 	$(KUSTOMIZE) build ../config/default > core-components.yaml
+	# Generate provider manifests.
+	# TODO: load the provider-version dynamically at rebase time when this is invoked by the Rebase Bot during one of its lifecycle hooks.
+	cd tools && $(MANIFESTS_GEN) --provider-name "cluster-api" --provider-type "CoreProvider" --provider-version "v1.5.2" --base-path "../../" --manifests-path "../manifests"
```

Comment:
Should we use a variable for now for this so that it can be overridden at runtime?

Reply:
Yes I was planning to deferred that until when we integrate the Cluster Bot workflow.
But it doesn't hurt to add that already.

Added in a requirement for an env var `PROVIDER_VERSION` being set.
Which is then used by `manifests-gen` to pull the right manifest version.